packages/components/src/components/TableComponent/TableComponent.js

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import {
+  Button,
+  DataTable,
+  DataTableSkeleton,
+  TableBatchAction,
+  TableBatchActions,
+  TableSelectAll,
+  TableSelectRow,
+  TableToolbar,
+  TableToolbarContent
+} from 'carbon-components-react';
+import Add from '@carbon/icons-react/lib/add/16';
+import Delete from '@carbon/icons-react/lib/delete/16';
+import { ALL_NAMESPACES } from '@tektoncd/dashboard-utils';
+
+import './TableComponent.scss';
+
+const {
+  TableContainer,
+  Table,
+  TableHead,
+  TableRow,
+  TableBody,
+  TableCell,
+  TableHeader
+} = DataTable;
+
+const TableComponent = props => {
+  const {
+    handleDelete,
+    handleNew,
+    loading,
+    dataHeaders,
+    dataRows,
+    selectedNamespace,
+    title
+  } = props;
+
+  return (
+    <div className="tableComponent">
+      <DataTable
+        useZebraStyles
+        rows={dataRows}
+        headers={dataHeaders}
+        isSortable
+        render={({
+          rows,
+          headers,
+          getHeaderProps,
+          getRowProps,
+          getSelectionProps,
+          getBatchActionProps,
+          selectedRows
+        }) => (
+          <TableContainer title={title}>
+            <TableToolbar>
+              <TableBatchActions {...getBatchActionProps()}>
+                {handleDelete && (
+                  <TableBatchAction
+                    id="delete-btn"
+                    data-testid="deleteButton"
+                    renderIcon={Delete}
+                    onClick={() => {
+                      handleDelete(
+                        selectedRows.map(secret => ({
+                          namespace: secret.id.split(':')[0],
+                          name: secret.id.split(':')[1]
+                        })),
+                        getBatchActionProps().onCancel
+                      );
+                    }}
+                  >
+                    Delete
```

Comment:
all strings should be externalised for translation using react-intl (`intl.formatMessage(...)` or the `<FormattedMessage {...} />` component)